src/core/client/framework/lib/network/customErrorMiddleware.ts

```diff
@@ -1,26 +1,12 @@
 import { Middleware } from "react-relay-network-modern/es";
-import { BadUserInputError, UnknownServerError } from "../errors";
 
-function getError(errors: Error[]): Error | null {
-  if (errors.length > 1 || !(errors[0] as any).extensions) {
-    // Multiple errors are GraphQL errors.
-    // TODO: (cvle) Is this assumption correct?
-    // No extensions == GraphQL error.
-    // TODO: (cvle) harmonize with server.
-    return null;
-  }
-  const err = errors[0];
-  if ((err as any).code === "BAD_USER_INPUT") {
-    return new BadUserInputError((err as any).extensions);
-  }
-  return new UnknownServerError(err.message, (err as any).extensions);
-}
+import extractError from "./extractError";
 
 const customErrorMiddleware: Middleware = next => async req => {
   const res = await next(req);
   if (req.isMutation() && res.errors) {
```

Comment:
Queries can also throw errors, is that something you need to change here to support?